roseWong/assignments/cyclicSort/lc268/lc268.py

```diff
@@ -0,0 +1,60 @@
+# Problem Statement #
+
+# We are given an array containing ‘n’ distinct numbers taken from the range 0 to ‘n’. Since the array has only ‘n’ numbers out of the total ‘n+1’ numbers, find the missing number.
+# given an array of length n and told that there is a missing number with 0 to n, we
+# need to sort the array, using the cyclic sort pattern, to find out the missing numbe
+# 
+
+def cyclic_sort(nums):
+  i = 0
+  total = 0
+  arr_total = 0
+  missing_num = 0
+  for i in range(len(nums)+1):
+    total = total + i
+  for i in range(len(nums)):
+    arr_total = arr_total + nums[i]
+  missing_num = total - arr_total
+  return missing_num
+
+# def find_missing_number(nums):
+#   # TODO: Write your code here
+#   return -1
+
+def main():
+  print(cyclic_sort([4, 0, 3, 1]))
+  print(cyclic_sort([8, 3, 5, 2, 4, 6, 0, 1]))
+
+# def main():
+#   print(find_missing_number([4, 0, 3, 1]))
+#   print(find_missing_number([8, 3, 5, 2, 4, 6, 0, 1]))
+main()
```

Comment:
I love this unique solution, but it does not match the line you wrote about using the cyclic sort.